addons/sale_project/tests/test_sale_project.py

```diff
@@ -518,6 +518,57 @@ def test_project_creation_on_so_confirm_with_account(self):
         self.assertEqual(sale_order.project_id[analytic_plan_name], sale_order.project_ids[analytic_plan_name], "The project created for the SO and the project of the SO should have the same account.")
         self.assertEqual(self.env.company, sale_order.project_ids.company_id, "The project created for the SO should have the same company as its account.")
 
+    def test_project_creation_on_so_with_manual_analytic(self):
+        """ Tests that the manually added analytic account (of a plan other than projects) and the project account
+            created when SO is confirmed are both still in the line after confirmation.
+        """
+        analytic_distribution_manual = {str(self.analytic_account_sale.id): 100}
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'partner_invoice_id': self.partner.id,
+            'partner_shipping_id': self.partner.id,
+            'order_line': [
+                Command.create({
+                    'product_id': self.product_order_service3.id,
+                    'analytic_distribution': analytic_distribution_manual,
+                }),
+            ],
+        })
+        self.assertEqual(sale_order.order_line.analytic_distribution, analytic_distribution_manual)
+        sale_order.action_confirm()
+        expected_analytic_distribution = analytic_distribution_manual | {str(sale_order.order_line.project_id.account_id.id): 100}
+        self.assertEqual(sale_order.order_line.analytic_distribution, expected_analytic_distribution)
+
+    def test_project_on_sol_with_analytic_distribution_model(self):
+        """ If a line has a distribution coming from an analytic distribution model, and the sale order has a project,
+            both the project account and the accounts from the ADM should still be in the line after confirmation.
+        """
+        distribution_model = self.env['account.analytic.distribution.model'].create({
+            'product_id': self.product_a.id,
+            'analytic_distribution': {self.analytic_account_sale.id: 100},
+            'company_id': self.company.id,
+        })
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'partner_invoice_id': self.partner.id,
+            'partner_shipping_id': self.partner.id,
+            'order_line': [
+                Command.create({'product_id': self.product_a.id}),
+            ],
+        })
+        self.assertEqual(sale_order.order_line.analytic_distribution, distribution_model.analytic_distribution)
+
+        project = self.env['project.project'].create({
+            'name': 'Project Test',
+            'account_id': self.analytic_account.id,
+            'allow_billable': True,
+        })
+        sale_order.project_id = project
```

Comment:
If I'm not mistaken, it won't work if you put `project_id` in the create vals because the compute function depends on itself (the computed value of `analytic_distribution` depends on the current value of `analytic_distribution`).
I believe this is an issue.